src/BaseValidator.php

```diff
@@ -35,15 +35,11 @@ public function __construct(
   }
 
   final public function getValidatedInput(): T {
-    if ($this->validated_input === null) {
-      if (!$this->validated) {
-        throw new \Exception(
-          'Must call `validate` before accessing validated input.',
-        );
-      }
-      throw new \Exception(
-        "Can't access validated input since validator is invalid.",
-      );
+    if (!$this->has_been_validated) {
```

Comment:
I think we could just do:

```
if (!$this->validated) {...}
if (!$this->isValid()) {...}
```
and then allow nulls?

It seems odd to me to track `is_valid` since `isValid` is doing the count of errors and `has_been_validated` is essentially the same as `validated`?